### PR TITLE
MODBULKOPS-3 - BulkOperationRepository

### PR DESCRIPTION
**Pull request by siarhei-charniak**
[MODBULKOPS-3](https://issues.folio.org/browse/MODBULKOPS-3) - BulkOperationRepository

## Purpose
BulkOperationRepository component should be implemented to cover CRUD operations for schemas from [https://issues.folio.org/browse/MODBULKOPS-2](https://issues.folio.org/browse/MODBULKOPS-2:)

## Approach
 * Created repositories
 * Created unit tests

## Pre-Merge Checklist:
Before merging this PR, please go through the following list and take appropriate actions.

- Does this PR meet or exceed the expected quality standards?
  - [x] Code coverage on new code is 80% or greater
  - [x] Duplications on new code is 3% or less
  - [x] There are no major code smells or security issues
- Does this introduce breaking changes?
  - [ ] Were any API paths or methods changed, added or removed?
  - [ ] Were there any schema changes?
  - [ ] Did any of the interface versions change?
  - [ ] Were permissions changed, added, or removed?
  - [ ] Are there new interface dependencies?
  - [x] There are no breaking changes in this PR.

If there are breaking changes, please **STOP** and consider the following:

- What other modules will these changes impact?
- Do JIRAs exist to update the impacted modules?
  - [ ] If not, please create them
  - [ ] Do they contain the appropriate level of detail?  Which endpoints/schemas changed, etc.
  - [ ] Do they have all they appropriate links to blocked/related issues?
- Are the JIRAs under active development?
  - [ ] If not, contact the project's PO and make sure they're aware of the urgency.
- Do PRs exist for these changes?
  - [ ] If so, have they been approved?

Ideally all of the PRs involved in breaking changes would be merged in the same day to avoid breaking the folio-testing environment.  Communication is paramount if that is to be achieved, especially as the number of intermodule and inter-team dependencies increase.

While it's helpful for reviewers to help identify potential problems, ensuring that it's safe to merge is ultimately the responsibility of the PR assignee.

### CODE REVIEW

**Review by alekGbuz — APPROVED** [review]

**Review by obozhko-folio — APPROVED** [review]

File: src/main/java/org/folio/bo/domain/entity/BulkOperation.java [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+import java.time.LocalDateTime;
+import java.util.UUID;
+
+@Data
+@Builder
+@With
+@AllArgsConstructor
+@NoArgsConstructor
+@Entity
+@Table(name = "bulk_operation")
```
Comment (khandramai) [inline src/main/java/org/folio/bo/domain/entity/BulkOperation.java:29]: Plural? bulk_operations?

File: src/main/java/org/folio/bo/domain/entity/BulkOperationDataProcessing.java [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+import java.time.LocalDateTime;
+import java.util.UUID;
+
+@Data
+@Builder
+@With
+@AllArgsConstructor
+@NoArgsConstructor
+@Entity
+@Table(name = "bulk_operation_data_processing")
```
Comment (khandramai) [inline src/main/java/org/folio/bo/domain/entity/BulkOperationDataProcessing.java:26]: Plural? bulk_operation_data_processings?

File: src/main/java/org/folio/bo/domain/entity/BulkOperationExecution.java [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+import java.time.LocalDateTime;
+import java.util.UUID;
+
+@Data
+@Builder
+@With
+@NoArgsConstructor
+@AllArgsConstructor
+@Entity
+@Table(name = "bulk_operation_execution")
```
Comment (khandramai) [inline src/main/java/org/folio/bo/domain/entity/BulkOperationExecution.java:26]: bulk_operation_execution?

File: src/main/java/org/folio/bo/domain/entity/BulkOperationExecutionChunk.java [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+import java.time.LocalDateTime;
+import java.util.UUID;
+
+@Data
+@Builder
+@With
+@AllArgsConstructor
+@NoArgsConstructor
+@Entity
+@Table(name = "bulk_operation_execution_chunk")
```
Comment (khandramai) [inline src/main/java/org/folio/bo/domain/entity/BulkOperationExecutionChunk.java:26]: bulk_operation_execution_chunks?

File: src/main/java/org/folio/bo/domain/entity/BulkOperationRule.java [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+import javax.persistence.Table;
+import java.util.UUID;
+
+@Data
+@Builder
+@With
+@AllArgsConstructor
+@NoArgsConstructor
+@Entity
+@Table(name = "bulk_operation_rule")
```
Comment (khandramai) [inline src/main/java/org/folio/bo/domain/entity/BulkOperationRule.java:25]: bulk_operation_rules?

File: src/main/java/org/folio/bo/domain/entity/BulkOperationExecutionContent.java [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+import javax.persistence.Table;
+import java.util.UUID;
+
+@Data
+@Builder
+@With
+@AllArgsConstructor
+@NoArgsConstructor
+@Entity
+@Table(name = "bulk_operation_execution_content")
```
Comment (khandramai) [inline src/main/java/org/folio/bo/domain/entity/BulkOperationExecutionContent.java:25]: bulk_operation_execution_contents?

**Comment by khandramai** [comment]
@siarhei-charniak , LGTM for me. The only question - is it makes sense to use table names in plural?

**Pull request merged**